src/states/abstractmps.jl

```diff
@@ -245,4 +234,93 @@ physicalspace(ψ::AbstractMPS) = map(Base.Fix1(physicalspace, ψ), eachsite(ψ))
 
 Return an iterator over the sites of the MPS `state`.
 """
-eachsite(ψ::AbstractMPS) = eachindex(ψ)
+eachsite(ψ::AbstractMPS) = eachsite(GeometryStyle(ψ), ψ)
+
+eachsite(::GeometryStyle, ψ::AbstractMPS) = eachindex(ψ)
```

Comment:
Is there a benefit to having this go through the geometrystyle? It looks like we end up with the same fallback anyways, so I don't really see the point here